### DISCUSSION TITLE
TileLayer with two sources at different zoom levels

### DISCUSSION THREAD

**Discussion by kylebarron** [visgl/deck.gl#4541] [Q&A]
The `TerrainLayer` specifically is in mind here, but this is a general `TileLayer` question: If you have two tiled datasets that depend on each other, is there a way to define their min and max zooms separately?

In a terrain use case, you might have a tiled terrain source that exists for zooms `1-13` but an imagery source that exists for zooms `1-15`. It might be possible to use separate `Tileset2D` instances for each data source?

**Comment by j6k4m8** [comment]
Probably the easiest way to do it is with two totally separate layers with different `minZoom` and `maxZoom` attributes...

**Reply by kylebarron** [reply]
> two tiled datasets that depend on each other

**Reply by j6k4m8** [reply]
As long as the coordinates overlap, I think it should still work, no?

**Reply by kylebarron** [reply]
No; in the original scenario, you might want to render a `TerrainLayer`, but your terrain data might only be valid up to zoom 12 while you have imagery up to zoom 15. I don't think there's currently a way to display the accurate imager data at zooms 12-15 while using terrain mesh from zoom 12. This is a scenario where the two data sources necessarily depend on each other.

To fix this, you'd need to allow each data source to "overzoom" independently. You'd in effect need to select a portion of the terrain source and resize it to your tile size, so that the terrain and image can use the same pixel coordinates in the GPU.

**Reply by charlieforward9** [reply]
Initiating this feature here https://github.com/visgl/deck.gl/pull/10296

Continuing conversation in issue & PR, closing
